ui/components/app/modals/customize-nonce/customize-nonce.test.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { fireEvent } from '@testing-library/react';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { renderWithProvider } from '../../../../../test/lib/render-helpers';
+import CustomizeNonce from '.';
+
+const mockHideModal = jest.fn();
+
+jest.mock('../../../../store/actions.ts', () => ({
+  ...jest.requireActual('../../../../store/actions.ts'),
+  hideModal: () => mockHideModal,
+}));
+
+describe('Customize Nonce', () => {
+  const mockState = {
+    appState: {
+      modal: {
+        modalState: {
+          props: {},
+        },
+      },
+    },
+  };
+
+  const mockStore = configureMockStore([thunk])(mockState);
+
+  const props = {
+    nextNonce: 1,
+    customNonceValue: '',
+    updateCustomNonce: jest.fn(),
+    getNextNonce: jest.fn(),
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should match snapshot', () => {
+    const { container } = renderWithProvider(
+      <CustomizeNonce {...props} />,
+      mockStore,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should change and submit custom nonce', async () => {
+    const { queryByTestId, queryByText } = renderWithProvider(
+      <CustomizeNonce {...props} />,
+      mockStore,
+    );
+
+    const nonceInputEvent = {
+      target: {
+        value: 101,
+      },
+    };
+
+    const nonceInput = queryByTestId('custom-nonce-input');
+    fireEvent.change(nonceInput, nonceInputEvent);
+    expect(nonceInput).toHaveValue(101);
+
+    const saveButton = queryByText('Save');
+    fireEvent.click(saveButton);
+
+    expect(props.updateCustomNonce).toHaveBeenCalledWith('101');
+    expect(props.getNextNonce).toHaveBeenCalled();
+    expect(mockHideModal).toHaveBeenCalled();
+  });
+
+  it('should handle emptry string custom nonce', () => {
+    const { queryByTestId, queryByText } = renderWithProvider(
+      <CustomizeNonce {...props} />,
+      mockStore,
+    );
+
+    const nonceInputEvent = {
+      target: {
+        value: '',
+      },
+    };
+
+    const nonceInput = queryByTestId('custom-nonce-input');
+    fireEvent.change(nonceInput, nonceInputEvent);
+
+    const saveButton = queryByText('Save');
+
+    fireEvent.click(saveButton);
+
+    expect(props.updateCustomNonce).toHaveBeenCalledWith(
+      props.customNonceValue,
+    );
+  });
+
+  it('should handle cancel', async () => {
+    const { queryByText } = renderWithProvider(
+      <CustomizeNonce {...props} />,
+      mockStore,
+    );
+
+    const modalClose = queryByText('Cancel');
+
+    fireEvent.click(modalClose);
+
+    expect(mockHideModal).toHaveBeenCalled();
```

Comment:
Question: this function should be tested in general <Modal /> component ideally ? we dont need to test again i think :) 
Check this out: https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/modal/modal.component.test.js#L28

Reply:
There is a case to be made for reducing redundancy here. At an elementary point, we would miss 100% coverage on the files. On another, what [modal.component.test.js#L28](https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/modal/modal.component.test.js#L28) tests that whatever function we pass to `onCancel` it will be called, not what actual func is being called. In this case, the `hideModal` from our actions is what is being called which is what we are testing for in this instance. 

Let's say in the future the func passed was changed to `exportAllPrivateKeys` when `onCancel` is called, `modal.component.test.js` would still pass, but that is not what is supposed to happen and the test should fail in that instance.